hydra/_internal/core_plugins/importlib_resources_config_source.py

```diff
@@ -40,23 +41,32 @@ def load_config(
         if not res.exists():
             raise ConfigLoadError(f"Config not found : {normalized_config_path}")
 
-        with res.open(encoding="utf-8") as f:
-            header_text = f.read(512)
-            header = ConfigSource._get_header_dict(header_text)
-            self._update_package_in_header(
-                header=header,
-                normalized_config_path=normalized_config_path,
-                is_primary_config=is_primary_config,
-                package_override=package_override,
-            )
-            f.seek(0)
-            cfg = OmegaConf.load(f)
-            return ConfigResult(
-                config=self._embed_config(cfg, header["package"]),
-                path=f"{self.scheme()}://{self.path}",
-                provider=self.provider,
-                header=header,
-            )
+        zipfile_path = sys.version_info[0:2] >= (3, 8) and isinstance(res, zipfile.Path)
+        if zipfile_path:
+            f = res.open()
+        else:
+            f = res.open(encoding="utf-8")
+
+        header_text = f.read(512)
+        if zipfile_path:
+            header_text = header_text.decode("utf-8")
+        header = ConfigSource._get_header_dict(header_text)
+        f.seek(0)
+        cfg = OmegaConf.load(f)
+        f.close()
```

Comment:
close in a finally block.
an exception between open and close and you wouldn't be closing the file at all.